yarn-project/simulator/src/avm/avm_machine_state.ts

```diff
@@ -52,6 +54,28 @@ export class AvmMachineState {
     return new AvmMachineState(state.l1GasLeft, state.l2GasLeft, state.daGasLeft);
   }
 
+  /**
+   * Consumes the given gas.
+   * Should any of the gas dimensions get depleted, it sets all gas left to zero and triggers
+   * an exceptional halt by throwing an OutOfGasError.
+   */
+  public consumeGas(gasCost: Partial<GasCost>) {
+    // Assert there is enough gas on every dimension.
+    const outOfGasDimensions = GasDimensions.filter(
+      dimension => this[`${dimension}Left`] - (gasCost[dimension] ?? 0) < 0,
```

Comment:
I think the YP says that it has to be strictly bigger than 0? It might be a mistake (I don't see why 0 would be wrong if you don't need to spend anything). Would you mind updating that in the YP? (I don't mind if you do it later).

Reply:
Hah, good catch!